controllers/datafoundationclient.go

```diff
@@ -248,3 +275,44 @@ func (r *dataFoundationClientReconciler) reconcileCSIKMSConnectionDetailsConfigM
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.list(storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := map[string]bool{}
+	for i := range storageClassList.Items {
+		storageClass := storageClassList.Items[i]
+		if storageClass.Provisioner == r.offering.Namespace+ocsCephFsProvisionerSuffix || storageClass.Provisioner == r.offering.Namespace+ocsRBDProvisionerSuffix {
+			ocsStorageClass[storageClass.Name] = true
+		}
+	}
+	pvList := &corev1.PersistentVolumeList{}
+	if err := r.list(pvList); err != nil {
```

Comment:
I believe that this will only list PVs from the set of namespaces the operator group allows for it. 
We need to list all PVs! for that I believe you can use the Unrestricted Client

Reply:
Updated to fetch using UnrestrictedClient